src/components/NewsList/NewsListItem.js

```diff
@@ -1,37 +1,26 @@
-import React, { Component } from 'react';
+import React from 'react';
 import { StyleSheet, TouchableOpacity, Text, View } from 'react-native';
 import PropTypes from 'prop-types';
 
-class NewsListItem extends Component {
-  constructor(props) {
-    super(props);
-    this.handleItemPress = this.handleItemPress.bind(this);
-  }
+export default function NewsListItem({ newsItem, style, onPress }) {
+  const { headline, source, snippet, web_url: url } = newsItem;
 
-  handleItemPress() {
-    const { newsItem, onPress } = this.props;
-    onPress(newsItem);
-  }
-
-  render() {
-    const { newsItem, style } = this.props;
-    const { headline, source, snippet } = newsItem;
-
-    return (
-      <TouchableOpacity
-        style={(styles.container, style)}
-        onPress={this.handleItemPress}
-      >
-        <View style={styles.headlineView}>
-          <Text style={styles.headlineText}>{headline.main}</Text>
-        </View>
-        <Text style={styles.snippetText}>{snippet}</Text>
-        <View style={styles.sourceView}>
-          <Text style={styles.sourceText}>{source}</Text>
-        </View>
-      </TouchableOpacity>
-    );
-  }
+  return (
+    <TouchableOpacity
+      style={(styles.container, style)}
+      onPress={() => {
```

Comment:
Don't use arrow functions like that, you need `this.handlePress` so it means you need to use class component instead of a functional component.